scene/gui/control.cpp

```diff
@@ -2724,6 +2724,29 @@ void Control::_window_find_focus_neighbor(const Vector2 &p_dir, Node *p_at, cons
 	}
 }
 
+void Control::set_autofocus(bool p_enable) {
+	if (data.autofocus == p_enable) {
+		return;
+	}
+	data.autofocus = p_enable;
+}
+
+bool Control::get_autofocus() const {
+	return data.autofocus;
+}
+
+void Control::attempt_autofocus() {
+#ifdef TOOLS_ENABLED
+	if (Engine::get_singleton()->is_editor_hint()) {
+		return;
+	}
+#endif
+
+	if (get_autofocus() && _is_focus_mode_enabled()) {
```

Comment:
`_is_focus_mode_enabled` is checking for recursive override only, it's not sufficient to check if control is focusable.

```suggestion
	bool ac_enabled = get_tree() && get_tree()->is_accessibility_enabled();
if (get_autofocus() && ((get_focus_mode_with_override() == FOCUS_ALL) || (get_focus_mode_with_override() == FOCUS_CLICK) || (ac_enabled && get_focus_mode_with_override() == FOCUS_ACCESSIBILITY))) {
```